### PR TITLE
🌱 fix TestUpdateCoreDNS flakes

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR attempts to remove flakes from the TestUpdateCoreDNS unit test by wrapping assertions in an Eventually loop

### CODE REVIEW

File: controlplane/kubeadm/internal/workload_cluster_coredns_test.go [hunk 1]
```diff
@@ -426,10 +426,21 @@ kind: ClusterConfiguration
 
 				// assert CoreDNS corefile
 				var expectedConfigMap corev1.ConfigMap
-				g.Expect(env.Get(ctx, client.ObjectKey{Name: coreDNSKey, Namespace: metav1.NamespaceSystem}, &expectedConfigMap)).To(Succeed())
-				g.Expect(expectedConfigMap.Data).To(HaveLen(2))
-				g.Expect(expectedConfigMap.Data).To(HaveKeyWithValue("Corefile", "updated-core-file"))
-				g.Expect(expectedConfigMap.Data).To(HaveKeyWithValue("Corefile-backup", expectedCorefile))
+				g.Eventually(func() error {
+					if err := env.Get(ctx, client.ObjectKey{Name: coreDNSKey, Namespace: metav1.NamespaceSystem}, &expectedConfigMap); err != nil {
```
Comment (enxebre) [inline controlplane/kubeadm/internal/workload_cluster_coredns_test.go:430]: Would it be more idiomatic with gomega if we do?:
```
g.Eventually(func() bool {

Expect(err).NotTo(HaveOccurred())
Expect(err).To(BeEquivalentTo(2))
...
return true
})
```

Comment (fabriziopandini) [reply controlplane/kubeadm/internal/workload_cluster_coredns_test.go:430]: AFAIK failing gomega assertion panics, and this doesn't gives us chances to handle errors properly, or in this case, to retry.
This popped out also in https://github.com/kubernetes-sigs/cluster-api/pull/4667#discussion_r638868901

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Pull request merged**